liblangutil/EVMVersion.h

```diff
@@ -106,7 +109,7 @@ class EVMVersion:
 	bool canOverchargeGasForCall() const { return *this >= tangerineWhistle(); }
 
 private:
-	enum class Version { Homestead, TangerineWhistle, SpuriousDragon, Byzantium, Constantinople, Petersburg, Istanbul, Berlin, London, Paris };
+	enum class Version { Homestead, TangerineWhistle, SpuriousDragon, Byzantium, Constantinople, Petersburg, Istanbul, Berlin, London, Paris, Shanghai };
 
 	EVMVersion(Version _version): m_version(_version) {}
```

Comment:
Shouldn't `m_version` already be set to `Version::Shanghai` here? (two lines below this comment) But maybe I missed some discussion about this. I was just wondering why the evm asm json import/export was still working ;)

@hrkrshnn @ekpyron

Reply:
Ok.. looks like that this PR seem to work different when I thought. I expected if I change `m_version` to `Version::Shanghai` the import/export evm asm tests would just always fail. I will look into this later.

Reply:
> Shouldn't m_version already be set to Version::Shanghai here? (two lines below this comment) But maybe I missed some discussion about this. I was just wondering why the evm asm json import/export was still working ;)

Still debating it. The default is currently left to Paris. I had a comment about it in the older commit: https://github.com/ethereum/solidity/pull/14107#discussion_r1162584329

Reply:
> Ok.. looks like that this PR seem to work different when I thought. I expected if I change m_version to Version::Shanghai the import/export evm asm tests would just always fail. I will look into this later.

Why? I think the current test suite should be run on all versions in one of the CI jobs, including shanghai. If you think there are tests that should fail, we have to investigate it.

Reply:
I was just thinking that in the context of https://github.com/ethereum/solidity/pull/13673. Somehow I expected that there need to be new code added for the asm json export - and for the import (to support `push0` as a new instruction). I did not have the time yet to read the code of your PR - I'm still a bit busy with other things. However, it looks like that you implemented it in a way that it just works. But yeah, I need to understand first how you did it ;) Right now I'm guessing that your code is just still compatible with `AssemblyItemType::Push`. But yeah, I will look into this soon :)

Reply:
ok, cool! `push0` is just treated as a normal `push` - somehow I expected here a special case. But I guess that it would not have any practical use. E.g. if someone really want to distinguish between `push1 0` and `push0` in the evm assembly, so both cases are there just `push 0` in the assembly.